examples/backports.pp

```diff
@@ -1 +1,32 @@
-# TODO
+$location = $::apt::distcodename ? {
+  'squeeze' => 'http://backports.debian.org/debian-backports',
+  'wheezy'  => 'http://ftp.debian.org/debian/',
+  default   => 'http://us.archive.ubuntu.com/ubuntu',
+}
+
+if $::apt::distid == 'debian' {
+  $repos   = 'main contrib non-free'
+  $key     = 'A1BD8E9D78F7FE5C3E65D8AF8B48AD6246925553'
+  $release = $::apt::distcodename
+} else {
+  $repos   = 'main universe multiverse restricted'
+  $key     = '630239CC130E1A7FD81A27B140976EAF437D05B5'
+  $release = $::apt::distcodename
+}
+
+# set up the actual backports
+apt::pin { 'backports':
+  before   => Apt::Source['backports'],
```

Comment:
As an aside, pin's are immediately reflected in apt, they do not require an `apt-get update` so we should alter `apt::pin` to not notify `apt-get update` as to not needlessly refresh things on a run.